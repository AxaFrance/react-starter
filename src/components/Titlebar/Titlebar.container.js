import React, { useEffect } from 'react';
import { Titlebar } from './Titlebar';

const EnhancedTitlebar = ({ title, subtitle }) => {
  useEffect(() => {}, []);

  return <Titlebar title={title} subtitle={subtitle} />;
};

export default EnhancedTitlebar;
