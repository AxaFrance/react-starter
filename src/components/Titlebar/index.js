export { default } from './Titlebar.container';
