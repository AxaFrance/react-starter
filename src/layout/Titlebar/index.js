export { Titlebar } from './Titlebar';
export { default } from './Titlebar.container';
