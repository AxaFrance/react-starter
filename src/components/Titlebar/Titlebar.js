import React from 'react';
import PropTypes from 'prop-types';
import './Titlebar.scss';

export const Titlebar = ({ title, subtitle }) => (
  <div className="af-title-bar">
    <div className="af-title-bar__wrapper container-fluid container">
      <h1 className="af-title-bar__title">
        {title}
        {subtitle && (
          <small className="af-title-bar__subtitle">{subtitle}</small>
        )}
      </h1>
    </div>
  </div>
);

const propTypes = {
  title: PropTypes.string.isRequired,
};

Titlebar.propTypes = propTypes;
