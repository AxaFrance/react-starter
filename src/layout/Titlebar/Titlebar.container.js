import React, { useEffect } from 'react';
import Titlebar from './Titlebar';

const EnhancedTitlebar = props => {
  useEffect(() => {}, []);

  return <Titlebar {...props} />;
};

export default EnhancedTitlebar;
