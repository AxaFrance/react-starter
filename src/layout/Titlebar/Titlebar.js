import React from 'react';
import PropTypes from 'prop-types';
import './Titlebar.scss';

export const Titlebar = ({ title }) => (
  <div className="af-title-bar">
    <div className="af-title-bar__wrapper container-fluid container">
      <h1 className="af-title-bar__title">{title}</h1>
    </div>
  </div>
);

const propTypes = {
  title: PropTypes.string.isRequired,
};

Titlebar.propTypes = propTypes;
