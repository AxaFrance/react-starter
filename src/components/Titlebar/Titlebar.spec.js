import React from 'react';
import { render } from '@testing-library/react';
import { Titlebar } from './Titlebar';

it('1. Renders Titlebar component without crashing', () => {
  const { asFragment } = render(<Titlebar title="test" />);
  expect(asFragment()).toMatchSnapshot();
});
