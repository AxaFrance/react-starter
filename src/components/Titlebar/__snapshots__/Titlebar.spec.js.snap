// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Renders Titlebar component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="af-title-bar__wrapper container-fluid container"
    >
      <h1
        class="af-title-bar__title"
      >
        test
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
